### PR TITLE
make Olive setup could be run without ORT installation

### PR DESCRIPTION
**Pull request by guotuofeng**
## Describe your changes
* refactor existing setup dependency logic to make sure the dml support be more easy to understand
* decouple olive setup with ORT.

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: olive/workflows/run/config.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 from olive.engine.packaging.packaging_config import PackagingConfig
 from olive.evaluator.olive_evaluator import OliveEvaluatorConfig
 from olive.model import ModelConfig
-from olive.passes import FullPassConfig, Pass
+from olive.passes import FullPassConfig
```
Comment (jambayk) [inline olive/workflows/run/config.py:19]: I don't think there is a need to make the import of `Pass` lazy? You already made `Pass` importable by making ort imports lazy. 

Even if `Pass` is not imported at top level, the validator `validate_pass_search` gets called anyways even in setup mode in `run.py:lines 210-213`

Comment (jambayk) [reply olive/workflows/run/config.py:19]: Also any imports of/from `olive.passes` will automatically trigger the `__init__` imports in all sub modules anyways

Comment (guotuofeng) [reply olive/workflows/run/config.py:19]: yes, you are right. will revert the change

File: olive/engine/engine.py [hunk 2]
```diff
@@ -23,13 +23,15 @@
...
 from olive.hardware import AcceleratorLookup, AcceleratorSpec, Device
 from olive.model import ModelConfig
-from olive.passes.olive_pass import Pass
 from olive.strategy.search_strategy import SearchStrategy
 from olive.systems.common import SystemType
 from olive.systems.local import LocalSystem
 from olive.systems.olive_system import OliveSystem
 from olive.systems.utils import create_new_system_with_cache
 
+if TYPE_CHECKING:
```
Comment (jambayk) [inline olive/engine/engine.py:32]: nit: are these changes also needed?

Comment (jambayk) [reply olive/engine/engine.py:32]: Or this method is recommended since we are only using `Pass` for type hints?

Comment (guotuofeng) [reply olive/engine/engine.py:32]: I add the TYPE_CHECKING here to make sure the explicit Pass import is optional for engine running.

Comment (jambayk) [reply olive/engine/engine.py:32]: Sounds good to me. Thanks!

Comment (guotuofeng) [reply olive/engine/engine.py:32]: as for best practice, any hint import should be put into TYPE_CHECKING.

**Review by jambayk — APPROVED** [review]

**Comment by jambayk** [comment]
Merging this since I have some changes that have merge conflicts with this.

**Pull request merged**